addons/purchase_requisition/wizard/purchase_requisition_alternative_warning.py

```diff
@@ -12,12 +13,29 @@ class PurchaseRequisitionAlternativeWarning(models.TransientModel):
     alternative_po_ids = fields.Many2many('purchase.order', 'warning_purchase_order_alternative_rel', string="Alternative POs")
 
     def action_keep_alternatives(self):
+        self._set_dest_moves()
         return self._action_done()
 
     def action_cancel_alternatives(self):
+        self._set_dest_moves()
         # in theory alternative_po_ids shouldn't have any po_ids in it, but it's possible by accident/forcing it, so avoid cancelling them to be safe
         self.alternative_po_ids.filtered(lambda po: po.state in ['draft', 'sent', 'to approve'] and po.id not in self.po_ids.ids).button_cancel()
         return self._action_done()
 
     def _action_done(self):
         return self.po_ids.with_context({'skip_alternative_check': True}).button_confirm()
+
+    def _set_dest_moves(self):
+        original_po = self.alternative_po_ids.filtered(lambda po: not po.partner_id)
```

Comment:
This expects very specific flow and is perhaps too hacky. Users can always set a vendor after the PO is created without a vendor or possible do something like purchase part of the original qty from one vendor + confirm => confirm an alternative PO for the rest of the qty => this linkage will fail because the warning won't pop up.

Also, if I understood your code earlier, you're requiring newly created alternative POs to have a vendor to avoid an error here, correct? If you avoid the check on the a po without an vendor, then you shouldn't need that restriction either. I think you should make something more simple, like: look `if PO.order_line.move_dest_ids`, then assume it's that one. That being said, this feels a bit too magic + will fail to work correctly in many use cases + the `move_dest_ids` are defined in `purchase_stock`, so you can't add this logic in this module because it doesn't depend on `stock`